### ISSUE TITLE
HelmCluster does not work when deployed in cluster that is not defined in current-context

### ISSUE THREAD

**Issue by sushi30** [dask/dask-kubernetes#321]
## Description

When deploying the dask helm chart in a namespace other than `default`, HelmCluster fails to initiate with `ConnectionError: kubectl port forward failed`.

## Minimal scenario:
In console

```console
helm repo add dask https://helm.dask.org
helm repo update
helm upgrade -i dask dask/dask --create-namespace -n dask
```
Then in jupyter:

```python
from dask_kubernetes import HelmCluster
cluster = HelmCluster("dask", namespace="dask")
```

```python
---------------------------------------------------------------------------
ConnectionError                           Traceback (most recent call last)
<ipython-input-31-06ad5cb9da3a> in <module>
----> 1 cluster = HelmCluster("dask", namespace="dask")

~/dev/notebooks/.venv/lib/python3.8/site-packages/dask_kubernetes/helm.py in __init__(self, release_name, auth, namespace, port_forward_cluster_ip, loop, asynchronous, scheduler_name, worker_name, node_host, node_port, **kwargs)
    118         if not self.asynchronous:
    119             self._loop_runner.start()
--> 120             self.sync(self._start)
    121 
    122     async def _start(self):

~/dev/notebooks/.venv/lib/python3.8/site-packages/distributed/deploy/cluster.py in sync(self, func, asynchronous, callback_timeout, *args, **kwargs)
    187             return future
    188         else:
--> 189             return sync(self.loop, func, *args, **kwargs)
    190 
    191     def _log(self, log):

~/dev/notebooks/.venv/lib/python3.8/site-packages/distributed/utils.py in sync(loop, func, callback_timeout, *args, **kwargs)
    349     if error[0]:
    350         typ, exc, tb = error[0]
--> 351         raise exc.with_traceback(tb)
    352     else:
    353         return result[0]

~/dev/notebooks/.venv/lib/python3.8/site-packages/distributed/utils.py in f()
    332             if callback_timeout is not None:
    333                 future = asyncio.wait_for(future, callback_timeout)
--> 334             result[0] = yield future
    335         except Exception as exc:
    336             error[0] = sys.exc_info()

~/dev/notebooks/.venv/lib/python3.8/site-packages/tornado/gen.py in run(self)
    760 
    761                     try:
--> 762                         value = future.result()
    763                     except Exception:
    764                         exc_info = sys.exc_info()

~/dev/notebooks/.venv/lib/python3.8/site-packages/dask_kubernetes/helm.py in _start(self)
    124         self.core_api = kubernetes.client.CoreV1Api()
    125         self.apps_api = kubernetes.client.AppsV1Api()
--> 126         self.scheduler_comm = rpc(await self._get_scheduler_address())
    127         await super()._start()
    128 

~/dev/notebooks/.venv/lib/python3.8/site-packages/dask_kubernetes/helm.py in _get_scheduler_address(self)
    154             service_name, self.namespace
    155         )
--> 156         address = await get_external_address_for_scheduler_service(
    157             self.core_api, service, port_forward_cluster_ip=self.port_forward_cluster_ip
    158         )

~/dev/notebooks/.venv/lib/python3.8/site-packages/dask_kubernetes/utils.py in get_external_address_for_scheduler_service(core_api, service, port_forward_cluster_ip)
     64             # If we are outside it will fail and we need to port forward the service.
     65             host = "localhost"
---> 66             port = await port_forward_service(service.metadata.name, port)
     67     return f"tcp://{host}:{port}"
     68 

~/dev/notebooks/.venv/lib/python3.8/site-packages/dask_kubernetes/utils.py in port_forward_service(service_name, remote_port, local_port)
     85     if await is_comm_open("localhost", local_port):
     86         return local_port
---> 87     raise ConnectionError("kubectl port forward failed")
     88 
     89 

ConnectionError: kubectl port forward failed
```

## Suggestion

This appears to be because of this line not specifying `--namespace` flag.
https://github.com/dask/dask-kubernetes/blob/8c8f7606b7fc754cd51782ca16bad103b0f252b7/dask_kubernetes/utils.py#L74-L81

## Workaround

By running:
```console
kubectl config set-context --current --namespace=dask
```
We can get the HelmCluster working. But this changes the kuberenetes context and is generally not considered good practice (by the virtue of never encountering it as a solution in any documentation).

**Comment by jacobtomlinson** [comment]
Nice catch, thanks for raising this.

I change my kubernetes context frequently (usually with [`kubectx`](https://github.com/ahmetb/kubectx) for ease). I wouldn't say this is bad practice, but it definitely isnt the *onyl* practice.

This is definitely a bug and we should add the flax to be explicit.

Do you have any interest in raising a Pull Request to fix this?

**Comment by jacobtomlinson** [comment]
Fixed in #325
